### DISCUSSION TITLE
Pixel streaming

### DISCUSSION THREAD

**Discussion by stellanhaglund** [bevyengine/bevy#5662] [Q&A]
Hi, I'm a little interested in implementing pixel streaming like UE has. 

Does anybody have a good idea of where to start, would rendering to a texture and passing that to the webrtc stream be enough or should it be deeper?

**Comment by mockersf** [comment]
Could you add more details to what "pixel streaming" is, and what you would want in Bevy?

**Comment by stellanhaglund** [comment]
Sure, Unreal has a plugin that enables pixel streaming.
Which is basically that you stream the window output through webrtc to a
web client, and the inputs from the webclient are forwarded to the running
application.
In unreal you can also choose if the application should be started headless
or normally.
I guess what’s happening is that they are passing the output of the gpu
pipeline to webrtc.
I was thinking that render to texture would be a good option but maybe
there is some better way

**Reply by stellanhaglund** [reply]
Here is a demo.

https://youtu.be/gkqZe5OMlj4

**Comment by cs50victor** [comment]
@stellanhaglund something like this: https://github.com/rlamarche/bevy_streaming ?
